vnext/Desktop/WinHTTPWebSocketResource.cpp

```diff
@@ -0,0 +1,447 @@
+
+WinHTTPWebSocketResource::~WinHTTPWebSocketResource() /*override*/
+{
+  if (m_webSocketHandle != NULL)
+  {
+    WinHttpCloseHandle(m_webSocketHandle);
+    m_webSocketHandle = NULL;
+  }
+
+  if (m_requestHandle != NULL)
+  {
+    WinHttpCloseHandle(m_requestHandle);
+    m_requestHandle = NULL;
+  }
+
+  if (m_connectionHandle != NULL)
+  {
+    WinHttpCloseHandle(m_connectionHandle);
+    m_connectionHandle = NULL;
+  }
+
+  if (m_sessionHandle != NULL)
+  {
+    WinHttpCloseHandle(m_sessionHandle);
+    m_sessionHandle = NULL;
+  }
+}
+
+#pragma region IWebSocketResource
+
+void WinHTTPWebSocketResource::Connect(const Protocols& protocols, const Options& options)
+{
+  m_sessionHandle = WinHttpOpen(
+    L"React Native Windows",              // pszAgentW
+    WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY,  // dwAccessType
+    NULL,                                 // pszProxyW
+    NULL,                                 // pszProxyBypassW
+    m_openSessionFlags                    // dwFlags
+  );
+
+  if (m_sessionHandle == NULL)
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error starting session", ErrorType::Connection });
+    }
+  }
+
+  m_connectionHandle = WinHttpConnect(
+    m_sessionHandle,    // hSession
+    m_url.lpszHostName, // pswzServerName
+    m_url.nPort,        // nServerPort
+    0                   // dwReserved
+  );
+
+  if (m_connectionHandle == NULL)
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error opening connection", ErrorType::Connection });
+    }
+  }
+
+  m_requestHandle = WinHttpOpenRequest(
+    m_connectionHandle, // hConnect
+    L"GET",             // pwszVerb
+    m_url.lpszUrlPath,  // pwszObjectName
+    NULL,               // pwszVersion
+    NULL,               // pwszReferrer
+    NULL,               // ppwszAcceptTypes
+    0                   // dwFlags
+  );
+
+  if (m_requestHandle == NULL)
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error starting request", ErrorType::Connection });
+    }
+  }
+
+  if (!WinHttpSetOption(m_requestHandle, WINHTTP_OPTION_CONTEXT_VALUE, this, sizeof(this)))
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error configuring request", ErrorType::Connection });
+    }
+  }
+
+  // Set WinHTTP callbacks
+  if (WinHttpSetStatusCallback(
+    m_requestHandle,                          // hInternet
+    Callback,                                 // lpfnInternetCallback
+    WINHTTP_CALLBACK_FLAG_ALL_NOTIFICATIONS,  // dwNotificationFlags
+    0                                         // dwReserved
+  ) == WINHTTP_INVALID_STATUS_CALLBACK)
+  {
+    m_errorStatus = GetLastError();
+    throw std::exception("Can not create instance. Invalid callback.");
+  }
+
+  BOOL fStatus = FALSE; //TODO: Make member?
+  fStatus = WinHttpSetOption(
```

Comment:
use builtin type bool (and false)

Reply:
Let's discuss offline.
MS Docs for these APIs use WinSDK-defined types.